### ISSUE TITLE
Hyperparameters of original cross-encoder/ms-marco-MiniLM series models?

### ISSUE THREAD

**Issue by Liwen-Hu3** [UKPLab/sentence-transformers#3505]
Hi,  

Thank you for releasing the `cross-encoder/ms-marco-MiniLM-L6-v2` model.  
I’m trying to reproduce or fine-tune this model for my own task, but I could not find the exact hyperparameter configuration that was used during training.  

- Could you please share the training hyperparameters (e.g., batch size, learning rate, epochs, optimizer, warmup steps)?  
- Are these hyperparameters consistent across all `cross-encoder/ms-marco-*` models, or do they vary between models?  
- For fine-tuning on a new task, do you have any recommended hyperparameter settings or best practices?  

Thanks a lot for your time and for making these models available!

**Comment by tomaarsen** [comment]
Hello!

For a bit of context, I took over the Sentence Transformers project in 2023, and this model is from before my time (~2021).

I see now that the links from the cross-encoder model cards are broken, apologies for that. I'll try to resolve that soon. The model cards state that the training scripts could be found here: https://github.com/UKPLab/sentence-transformers/tree/v2.5-release/examples/training/ms_marco, which contained 2 cross-encoder training scripts:
* https://github.com/UKPLab/sentence-transformers/blob/v2.5-release/examples/training/ms_marco/train_cross-encoder_scratch.py train "from scratch"
* https://github.com/UKPLab/sentence-transformers/blob/v2.5-release/examples/training/ms_marco/train_cross-encoder_kd.py train with knowledge distillation

You can gather some hyperparameters from these as well.

These use the pre-v4 training style (`model.fit`) instead of the newer style (`SentenceTransformerTrainer`), but in v4.0 I updated the `model.fit` method to use the `SentenceTransformerTrainer` behind the scenes, with identical performance. In short: the old scripts will still work with newer versions of Sentence Transformers, even all the way up to v5.1.

However, in the v4.0 update, I also created refactored/modernized versions of these scripts:
* https://github.com/UKPLab/sentence-transformers/blob/master/examples/cross_encoder/training/ms_marco/training_ms_marco_bce_preprocessed.py uses the exact same dataset as the old "from scratch" script
* https://github.com/UKPLab/sentence-transformers/blob/master/examples/cross_encoder/training/distillation/train_cross_encoder_kd_mse.py uses the exact same dataset as the old knowledge distillation script

For https://huggingface.co/cross-encoder/ms-marco-MiniLM-L6-v2, I actually don't know with certainty how it was trained. In the [config](https://huggingface.co/cross-encoder/ms-marco-MiniLM-L6-v2/blob/main/config.json#L2) I can see that the base model is the [L12 model](https://huggingface.co/cross-encoder/ms-marco-MiniLM-L12-v2) instead, so perhaps it was trained with distillation to match the larger model. That model uses `microsoft/MiniLM-L12-H384-uncased`, so it was probably trained like one of the two scripts linked.

My personal recommendation is to read my [train-reranker blogpost](https://huggingface.co/blog/train-reranker), which uses the `BinaryCrossEntropy` loss (like the "from scratch") to train a model on the [GooAQ dataset](https://huggingface.co/datasets/sentence-transformers/gooaq). It has a lot of learnings about the various components, with some tips here and there. It also describes hard negatives mining briefly, which should help a lot as well.

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/885b8eb7-455e-41b1-994a-ab5f6d03e534" />

- Tom Aarsen

**Comment by Liwen-Hu3** [comment]
Thanks for the helpful response!

I’m fine-tuning MiniLM-L6 on a binary relevance task (1 = relevant, 0 = irrelevant) with BCE loss using my own dataset (million-level data). I’ve noticed the model quickly gets overconfident—metrics like F1, AUROC, and AUPRC go up, but eval loss also rises.

I tried 0.1 label smoothing, but it didn’t help much. Did you run into this too, and do you have any tips?
